pyramid_swagger/load_schema.py

```diff
@@ -160,49 +139,85 @@ def get_model_resolver(schema):
     return RefResolver('', '', models)
 
 
-class SchemaMap(namedtuple(
-        'SchemaMap', [
-            'request_query_schema',
-            'request_path_schema',
-            'request_body_schema',
-            'response_body_schema'
-        ])):
+class ValidatorMap(namedtuple('_VMap', 'query path headers body response')):
     """
-    A SchemaMap contains a mapping from incoming paths to schemas for request
-    queries, request bodies, and responses. This requires some precomputation
-    but means we can do fast query-time validation without having to walk over
-    the schema.
+    A data object with validators for each part of the request and response
+    objects. Each field is a :class:`SchemaValidator`.
     """
     __slots__ = ()
 
-
-def build_request_to_schemas_map(schema):
-    """Take the swagger schema and build a map from incoming path to a
-    jsonschema for requests and responses."""
-    request_to_schema = {}
+    @classmethod
+    def from_operation(cls, operation, models, resolver):
+        args = []
+        for schema, validator in [
+            (build_param_schema(operation, 'query'), Draft3Validator),
+            (build_param_schema(operation, 'path'), Draft3Validator),
+            (build_param_schema(operation, 'header'), Draft3Validator),
+            (extract_body_schema(operation, models), Draft4Validator),
+            (extract_response_body_schema(operation, models),
+                Draft4Validator),
+        ]:
+            args.append(SchemaValidator.from_schema(
+                schema,
+                resolver,
+                validator))
+
+        return cls(*args)
+
+
+class SchemaValidator(object):
+
+    def __init__(self, schema, validator):
+        self.schema = schema
+        self.validator = validator
+
+    @classmethod
+    def from_schema(cls, schema, resolver, validator_class):
+        return cls(
+            schema,
+            validator_class(schema, resolver=resolver, types=EXTENDED_TYPES))
+
+    def validate(self, values):
```

Comment:
This is the case when an operation doesn't define any params for the paramType.  We end up with a None schema from `build_param_schema`, so this just skips the validation since we don't have anything to validate.